Packages/blueberry-jam-core/Runtime/LevelLoader/SceneTransitionerHelper.cs

```diff
@@ -0,0 +1,90 @@
+using System.Collections;
+using Codice.Client.BaseCommands.TubeClient;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace BJ
+{
+    internal class SceneTransitionerHelper : SingletonGameObject<SceneTransitionerHelper>
+    {
+        [Header("Game Object References")]
+        [SerializeField] private CanvasGroup Crossfade;
+        [SerializeField] private Image LoadingIcon;
+        [Header("Animation Reference Storage")]
+        [SerializeField] private Sprite[] Frames;
+
+        [Header("Animation Configuration")]
+        [SerializeField] private float TransitionTime = 1f;
+        [SerializeField] private AnimationCurve EaseTransitionCurve;
+
+        /* ----- Helper Variables -----*/
+        private float StartTransitionTime;
+
+        protected override void Awake()
+        {
+            base.Awake();
+            Assertions();
+
+            StartTransitionTime = 0.0f;
```

Comment:
I may be letting my C++ habits get the better of me here.

I would like to confirm that doing this is equivalent to me substituting line 21 with `private float StartTransitionTime = 0.0f`. Am I correct?

Reply:
Strictly speaking no, practically yes.
If you do the line 21 substitution the initialization will happen when the class is instantiated.
As it is, this assignment operation happens in Awake. There is a small window between initialization and Awake where someone could conceivably set StartTransitionTime to something else, then Awake would override it. But this would be a really awkward way to do things in Unity so they're as good as the same.

Reply:
Sounds good, since the window is so small, I will leave this as is for now.